### PR TITLE
Feat/16187/text housekeeping

### LINKED ISSUE #16187: 🧹 [UI House Keeping] Text
### Description
Ensure that `Text` adheres to all of the following conventions and standards

- [ ] Has a `className` prop and the PropType descriptions are all the same
- [ ] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [ ] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [ ] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [ ] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [ ] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [ ] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [ ] All Base component MDX documentation have the base component notification at the top
- [ ] Add `mm-` prefix to all classNames
- [ ] className is kebab case version of the component name
- [ ] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [ ] Remove `Component.displayName` not needed
- [ ] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [ ] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [ ] Add snapshot testing

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

Text Component Housekeeping 🧹

Fixes: #16187 

### Before
Screen Shot 2022-11-17 at 3 44 32 PM

### After
Screen Shot 2022-11-17 at 3 44 14 PM

## Manual Testing Steps

`yarn test:unit:jest ui/components/component-library`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### Description
Ensure that `Text` adheres to all of the following conventions and standards

Text component specific
- [x] Remove default export 
- [ ] ~Change `body-lg-medium` to `body-lg` in variants~
- [x] Add `body-sm-bold` and `body-md-bold` to keep consistent pattern matching the `body-lg-medium`

General house keeping
- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [x] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [ ] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [ ] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [ ] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [ ] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [x] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### CODE REVIEW

File: ui/helpers/constants/design-system.js [hunk 1]
```diff
@@ -157,9 +157,11 @@ export const TEXT = {
...
   HEADING_SM: 'heading-sm',
-  BODY_LG: 'body-lg-medium',
+  BODY_LG_MEDIUM: 'body-lg-medium',
   BODY_MD: 'body-md',
+  BODY_MD_BOLD: 'body-md-bold',
   BODY_SM: 'body-sm',
+  BODY_SM_BOLD: 'body-sm-bold',
   BODY_XS: 'body-xs',
   INHERIT: 'inherit',
 };
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.js:167]: Does it make sense to move this constant to the component folder?

**Review by georgewrmarshall — COMMENTED** [review]
Looking good just have one question regarding TEXT const

File: ui/components/component-library/text/text.js [hunk 2]
```diff
@@ -98,7 +98,8 @@ export const Text = React.forwardRef(
 
 Text.propTypes = {
   /**
-   * The variation of font styles including sizes and weights of the Text component (display, heading, body)
+   * The variation of font styles including sizes and weights of the Text component
+   * Possible values: `DISPLAY_MD` 48px/32px,`HEADING_LG` 32px/24px,`HEADING_MD` 24px/18px,`HEADING_SM` 18px/16px,`BODY_LG_MEDIUM` 18px/16px,`BODY_MD` 16px/14px,`BODY_MD_BOLD` 16px/14px,`BODY_SM` 14px/12px,`BODY_SM_BOLD` 14px/12px,`BODY_XS` 12px/10px,`INHERIT`
```
Comment (georgewrmarshall) [inline ui/components/component-library/text/text.js:102]: Could we simplify this slightly by just including the font-size? Just wondering if line-height is a value that engineers use frequently?

File: ui/components/component-library/text/text.constants.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  HEADING_MD: 'heading-md',
+  HEADING_SM: 'heading-sm',
+  BODY_LG_MEDIUM: 'body-lg-medium',
+  BODY_MD: 'body-md',
+  BODY_MD_BOLD: 'body-md-bold',
+  BODY_SM: 'body-sm',
+  BODY_SM_BOLD: 'body-sm-bold',
+  BODY_XS: 'body-xs',
+  INHERIT: 'inherit',
+};
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/text/text.constants.js:19]: Hmm yeah I can't choose either. I like your suggestion would it be weird if we exported it from both?

File: ui/components/component-library/text/text.js [hunk 4]
```diff
@@ -98,7 +98,19 @@ export const Text = React.forwardRef(
...
+   * `DISPLAY_MD` LG Screen: 48px / Sm Screen: 32px,
+   * `HEADING_LG` LG Screen: 32px / Sm Screen: 24px,
+   * `HEADING_MD` LG Screen: 24px / Sm Screen: 18px,
+   * `HEADING_SM` LG Screen: 18px / Sm Screen: 16px,
+   * `BODY_LG_MEDIUM` LG Screen: 18px / Sm Screen: 16px,
+   * `BODY_MD` LG Screen: 16px / Sm Screen: 14px,
+   * `BODY_MD_BOLD` LG Screen: 16px / Sm Screen: 14px,
+   * `BODY_SM` LG Screen: 14px / Sm Screen: 12px,
+   * `BODY_SM_BOLD` LG Screen: 14px / Sm Screen: 12px,
+   * `BODY_XS` LG Screen: 12px / Sm Screen: 10px,
```
Comment (georgewrmarshall) [inline ui/components/component-library/text/text.js:112]: Can we weave in some familiar formatting using the screen size mixins?

```suggestion
   * `DISPLAY_MD` base font-size: 32px / screen-md-min { font-size: 48px }
   * `HEADING_LG` base font-size: 24px / screen-md-min { font-size: 32px  }
   * `HEADING_MD` base font-size: 18px / screen-md-min { font-size: 24px  }
   * `HEADING_SM` base font-size: 16px / screen-md-min { font-size: 18px  }
   * `BODY_LG_MEDIUM` base font-size: 16px / screen-md-min { font-size: 18px  }
   * `BODY_MD` base font-size: 14px / screen-md-min { font-size: 16px  }
   * `BODY_MD_BOLD` base font-size: 14px / screen-md-min { font-size: 16px  }
   * `BODY_SM` base font-size: 12px / screen-md-min { font-size: 14px  }
   * `BODY_SM_BOLD` base font-size: 12px / screen-md-min { font-size: 14px  }
   * `BODY_XS` base font-size: 10px / screen-md-min { font-size: 12px  }
```

Comment (georgewrmarshall) [reply ui/components/component-library/text/text.js:112]: No not sure. Was just floating an idea. From the inspect view there is nothing to indicate Lg Screen / Sm Screen so was trying to tie it to something semi familiar like the breakpoint mixin

Screen Shot 2022-11-22 at 10 01 55 AM

File: ui/components/component-library/text/text.js [hunk 5]
```diff
@@ -35,7 +35,7 @@ export const ValidTags = [
 export const Text = React.forwardRef(
   (
     {
-      variant = TEXT.BODY_MD,
+      variant = TEXT_VARIANTS.BODY_MD,
```
Comment (georgewrmarshall) [inline ui/components/component-library/text/text.js:38]: As per our convention. Use general design system constants unless for storybook controls or propTypes

```suggestion
      variant = TEXT.BODY_MD,
```

File: ui/components/component-library/text/text.stories.js [hunk 6]
```diff
@@ -6,9 +6,9 @@ import {
   BORDER_COLORS,
   FONT_WEIGHT,
   FONT_STYLE,
+  TEXT,
   TEXT_COLORS,
   TEXT_ALIGN,
-  TEXT,
   OVERFLOW_WRAP,
   TEXT_TRANSFORM,
   FRACTIONS,
```
Comment (georgewrmarshall) [inline ui/components/component-library/text/text.stories.js:14]: Any reason for this rearrange of imports?

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left some small suggestions and questions

**Review by georgewrmarshall — DISMISSED** [review]
LGTM 🚀

**Comment by georgewrmarshall** [comment]
Ah looks like merging https://github.com/MetaMask/metamask-extension/pull/16497 broke something

**Review by georgewrmarshall — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**